MineSweeper.py

```diff
@@ -0,0 +1,256 @@
+    mouse_click.append(event.y)
+
+    box_index_x = (mouse_click[0] - 10) // BOX_SIZE
+    box_index_y = (mouse_click[1] - 10) // BOX_SIZE
+
+    try:
+      board.__tiles[box_index_x][box_index_y].marked_mine = True
+
+    except IndexError:
+      pass
+
+    self.draw(game.panel)
+
+
+class Tile():
+
+  def __init__(self, x, y, size):
+    self.__x_index = x
+    self.__y_index = y
+    self.__x = x + 10
+    self.__y = y  + 10
+    self.__size = size
+
+    self.is_clicked = False
+    self.is_mine = False
+    self.marked_mine = False
+    self.value = 0
+
+  # Tile function to draw itself to the window. 
+  def draw(self, panel):
+
+    # If the tile has yet to be clicked, draw a dark gray box on top of a light gray box
+    # If the tile is marked as a mine, draw a cute lil flag on there
+    if(self.is_clicked == False):
+      panel.create_rectangle(self.__x, self.__y, self.__x + self.__size, 
+			     self.__y + self.__size, fill = 'grey')
+
+      panel.create_line(self.__x, self.__y, self.__x + self.__size, 
+			self.__y + self.__size, fill = 'dim grey')
+
+      panel.create_line(self.__x, self.__y + self.__size, self.__x + self.__size,
+			self.__y, fill = 'dim grey')
+
+      panel.create_rectangle(self.__x + 2, self.__y + 2, self.__x + self.__size - 2, 
+			     self.__y + self.__size - 2, fill = 'dim grey', outline = 'dim grey')
+
+      if(self.marked_mine):
+        panel.create_rectangle(self.__x + 5, self.__y + 5, self.__x + self.__size - 5, 
+			       self.__y + self.__size/2, fill = 'black') 
+
+        panel.create_line(self.__x + 5, self.__y + 5, self.__x + 5, 
+			  self.__y + self.__size - 2)
+
+    else:
+      if(self.is_mine):
+        panel.create_rectangle(self.__x, self.__y, self.__x + self.__size, 
+			       self.__y + self.__size, fill = 'grey')
+        panel.create_line(self.__x, self.__y, self.__x + self.__size, 
+			  self.__y + self.__size, fill = 'red')
+        panel.create_line(self.__x, self.__y + self.__size, self.__x + self.__size,
+			  self.__y, fill = 'red')
+        panel.create_line(self.__x, self.__y + self.__size/2, self.__x + self.__size, 
+			  self.__y + self.__size/2, fill = 'red')
+        panel.create_line(self.__x + self.__size/2, self.__y, self.__x + self.__size/2, 
+			  self.__y + self.__size, fill = 'red')
+      
+      else:
+        panel.create_rectangle(self.__x, self.__y, self.__x + self.__size, 
+			       self.__y + self.__size, fill = 'grey')
+
+        if(self.value != 0):
+          panel.create_text(self.__x + self.__size/2, self.__y + self.__size/2,
+			         text = str(self.value))
+
+  # Tile function to reveal a tile. 
+  def show_self(self):
+    self.is_clicked = True
+    self.marked_mine = False
+
+
+def main():
+
+  global mines, game, board
+
+  game = Game()
+
+  board = Board(game)
+  board.assign_mines()
+
+  game.master.bind("<Button-1>", board.left_click_event)
+  game.master.bind("<Button-2>", board.right_click_event)
```

Comment:
Ah, also because the function they are binding to is a method within the board class and I was getting an error that the global variable board wasn't found